packages/payment/src/models/payment-collection.ts

```diff
@@ -0,0 +1,147 @@
+import {
+  BeforeCreate,
+  Collection,
+  Entity,
+  Enum,
+  Filter,
+  ManyToMany,
+  OneToMany,
+  OnInit,
+  OptionalProps,
+  PrimaryKey,
+  Property,
+} from "@mikro-orm/core"
+import { DAL } from "@medusajs/types"
+
+import { DALUtils, generateEntityId } from "@medusajs/utils"
+import PaymentProvider from "./payment-provider"
+import PaymentSession from "./payment-session"
+import Payment from "./payment"
+
+/**
+ * @enum
+ *
+ * The payment collection's status.
+ */
+export enum PaymentCollectionStatus {
+  /**
+   * The payment collection isn't paid.
+   */
+  NOT_PAID = "not_paid",
+  /**
+   * The payment collection is awaiting payment.
+   */
+  AWAITING = "awaiting",
+  /**
+   * The payment collection is authorized.
+   */
+  AUTHORIZED = "authorized",
+  /**
+   * Some of the payments in the payment collection are authorized.
+   */
+  PARTIALLY_AUTHORIZED = "partially_authorized",
+  /**
+   * The payment collection is canceled.
+   */
+  CANCELED = "canceled",
+}
+
+type OptionalPaymentCollectionProps =
+  | "currency_code"
+  | "authorized_amount"
+  | "refunded_amount"
+  | "region_id"
+  | "completed_at"
+  | DAL.SoftDeletableEntityDateColumns
+
+@Entity({ tableName: "payment-collection" })
```

Comment:
todo: _snake_casify_
```suggestion
@Entity({ tableName: "payment_collection" })
```